fla/layers/mla.py

```
@@ -0,0 +1,219 @@
+from fla.modules import RMSNorm, RotaryEmbedding
+from transformers.utils import logging
+from typing import TYPE_CHECKING, Optional, Tuple
+from functools import partial
+from fla.ops.utils.index import prepare_lens_from_mask
+from fla.layers.utils import pad_input, unpad_input
+
+
+try:
+    from flash_attn import flash_attn_func, flash_attn_varlen_func
+except ImportError:
+    warnings.warn(
```

Ruff (0.8.2)

No explicit `stacklevel` keyword argument found

(B028)